key_gen.go

```diff
@@ -83,6 +101,24 @@ func KeyForPrefixedStringMap(
 	return key
 }
 
+func dedupeSortedStrings(in []string) (out []string) {
+	if len(in) == 0 {
+		return in
+	}
+
+	prev := in[0]
+	out = in[:1] // zero-alloc filtering
+	for _, v := range in[1:] {
+		if prev == v {
+			continue
```

Comment:
separate method is good for breaking up the logic of the above function, but this could probably be inlined into the for loop around keys. that way we avoid iterating over the same slice twice, and it's not much additional code (just keep `prev` and the check against `prev`)

Reply:
Good call. Switching.